tgui/packages/tgui/interfaces/RiotGear.js

```diff
@@ -0,0 +1,211 @@
+import { useBackend, useLocalState } from '../backend';
+import { Box, Button, Flex, Modal, Section, LabeledControls, TimeDisplay } from '../components';
+import { formatTime } from '../format';
+import { Window } from '../layouts';
+
+export const RiotGear = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    authed,
+    authorisations,
+    hasID,
+    hasAccess,
+    mismatchedAppearance,
+    mismatchedID,
+    maxSecAccess,
+    existingAuthorisation,
+    nameOnFile,
+    printOnFile,
+    cooldown,
+  } = data;
+
+  const [nameModal, displayNameModal] = useLocalState(context, 'nameIsOnFile', '');
+  const [printModal, displayPrintModal] = useLocalState(context, 'printIsOnFile', '');
+
+  return (
+    <Window
+      title={`Armoury Authorisation`}
+      width={800}
+      height={280}
+    >
+      <Window.Content>
+        {(!hasID && !authed) && (
```

Comment:
Since all these are the same component with different text, you can just move the logic chains inside the component to change what the text says to avoid redefining the component every time.